packages/world/ts/config/resolveWorldConfig.ts

```diff
@@ -17,7 +17,7 @@ export function resolveWorldConfig(
 ) {
   // Include contract names ending in "System", but not the base "System" contract, and not Interfaces
   const defaultSystemNames =
-    existingContracts?.filter((name) => name.endsWith("System") && name !== "System" && !name.match(/^I[A-Z]/)) ?? [];
+    existingContracts?.filter((name) => name.match(/^(?!I[A-Z]).*System$/) && !name.match(/^.+__I.+System$/)) ?? [];
```

Comment:
I think `(^|__)` is what you're looking for? "start of string or prefixed by __"

```suggestion
    existingContracts?.filter((name) => !/(^|___)I.+System$/.test(name)) ?? [];
```